From 395d7098cff931e9e6ef07c9d0eace5aa47bf864 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Wed, 17 Apr 2024 00:35:05 +0300
Subject: [PATCH 50/50] Initialize extra before calling
 unit_assign_specific_activity_target()

If the extra going in is not NULL, it's assumed to be ok to begin with,
and no real extra gets assigned.

See RM #460

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/actiontools.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/actiontools.c b/server/actiontools.c
index 590a46a215..2f6da9ade6 100644
--- a/server/actiontools.c
+++ b/server/actiontools.c
@@ -906,7 +906,7 @@ int action_sub_target_id_for_action(const struct action *paction,
   case ASTK_EXTRA_NOT_THERE:
     if (action_has_result(paction, ACTRES_PILLAGE)) {
       /* Special treatment for "Pillage" */
-      struct extra_type *pextra;
+      struct extra_type *pextra = NULL;
       enum unit_activity activity = action_get_activity(paction);
 
       unit_assign_specific_activity_target(actor_unit, &activity, &pextra);
-- 
2.43.0

